### PR TITLE
Take profile from parameters and use the default configuration path

### PR DESCRIPTION
**Pull request by julienw**

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -198,12 +210,27 @@ pub struct OpenzwaveAdapter {
...
-    pub fn init<T: AdapterManagerHandle + Send + Sync + 'static> (box_manager: &Arc<T>) -> Result<(), Error> {
+    pub fn init<T: AdapterManagerHandle + Send + Sync + 'static> (box_manager: &Arc<T>, config_path: &str) -> Result<(), Error> {
+
+        try!(ensure_directory(config_path));
+
         let options = InitOptions {
             device: None, // TODO we should expose this as a Value
-            config_path: "./config/openzwave/",
-            user_path: "./config/openzwave/",
+            config_path: ConfigPath::Default,
```
Comment (julienw) [inline src/lib.rs]: This needs https://github.com/fxbox/openzwave-stateful-rust/pull/14

**Comment by julienw** [comment]
You'll need https://github.com/fxbox/openzwave-stateful-rust/pull/14 to build, and https://github.com/fxbox/openzwave-rust/pull/26 to make it work properly.

**Comment by julienw** [comment]
@ferjm r?

File: src/lib.rs [hunk 2]
```diff
@@ -198,12 +210,30 @@ pub struct OpenzwaveAdapter {
...
+    if !path.exists() {
+        try!(fs::create_dir(path));
+    }
+
+    Ok(())
+}
+
 impl OpenzwaveAdapter {
-    pub fn init<T: AdapterManagerHandle + Send + Sync + 'static> (box_manager: &Arc<T>) -> Result<(), Error> {
+    pub fn init<T: AdapterManagerHandle + Send + Sync + 'static> (box_manager: &Arc<T>, config_path: &str) -> Result<(), Error> {
```
Comment (ferjm) [inline src/lib.rs]: Maybe s/config_path/user_path

File: src/lib.rs [hunk 3]
```diff
@@ -198,12 +210,30 @@ pub struct OpenzwaveAdapter {
...
+
+        try!(ensure_directory(config_path));
+
         let options = InitOptions {
             device: None, // TODO we should expose this as a Value
-            config_path: "./config/openzwave/",
-            user_path: "./config/openzwave/",
+            config_path: ConfigPath::Default,
+            user_path: config_path,
```
Comment (ferjm) [inline src/lib.rs]: Same as above. s/config_path/user_path ?

Comment (ferjm) [inline src/lib.rs]: Also, we could use some comments about what `config_path` and `user_path` are used for.

**Comment by ferjm** [comment]
Looks good. Thanks!

**Pull request merged**